### PR TITLE
Category promo

### PR DESCRIPTION
**Pull request by asmega**
- This allows the cms to add promoted items to a category

### CODE REVIEW

File: spec/models/category_promo_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+describe CategoryPromo do
+  it 'associations' do
```
Comment (neoeno) [inline spec/models/category_promo_spec.rb]: we've got this in sentence format [here](https://github.com/moneyadviceservice/cms/pull/227/files#diff-5cfe7e4f7bab0f6559c519ec3aa95b18R29), but you could equally use `specify 'associations' do`

File: app/models/category_promo.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class CategoryPromo < ActiveRecord::Base
+  PROMO_TYPES = %w{ blog recommended popular calculator tool }
+
+  belongs_to :category, class: Comfy::Cms::Category
+end
```
Comment (urfolomeus) [inline app/models/category_promo.rb]: I know your protecting against submissions with no title but shouldn't that be backed up with validations in here?

Same for ensuring promo type is included in PROMO_TYPES.

**Comment by neoeno** [comment]
:+1: from me, though @darokel makes a good point

File: spec/models/category_promo_spec.rb (hunk 1 above)
Comment (neoeno) [inline spec/models/category_promo_spec.rb]: Or `it { is_expected.to belong_to(:category) }`

File: app/models/category_promo.rb (hunk 2 above)
Comment (urfolomeus) [inline app/models/category_promo.rb]: Also for locale?

File: spec/models/category_promo_spec.rb (hunk 1 above)
Comment (asmega) [inline spec/models/category_promo_spec.rb]: The latter option looks good

File: app/models/category_promo.rb (hunk 2 above)
Comment (asmega) [inline app/models/category_promo.rb]: Agreed. I've now added these validations

**Comment by tomas-stefano** [comment]
:+1:

**Comment by darokel** [comment]
:+1: :ship:

**Comment by urfolomeus** [comment]
:+1:

**Pull request merged**